### ISSUE TITLE
E202 (no leading zero for octal permission) False positive

### ISSUE THREAD

**Issue by sourcejedi** [ansible/ansible-lint#461]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

- ansible-lint installation method: Ansible Galaxy use of ansible-lint

# Desired Behaviour

ansible-lint provides correct reason to following the lint

# Actual Behaviour (Bug report only)

Please give some details of what is actually happening.
Include a [minimum complete verifiable example](http://stackoverflow.com/help/mcve)
with:
- playbook
- output of running ansible-lint
- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`

"Numeric file permissions without leading zero can behave in unexpected ways. See http://docs.ansible.com/ansible/file_module.html" - https://docs.ansible.com/ansible-lint/rules/default_rules.html

./galaxy/downloads/tmp2REDBx/tasks/main-tasks.yml:4: [E202] Octal file permissions must contain leading zero

```
# Lets create the configuration first...
# avoid locking ourself out.

- name: Create firewalld config directories
  file:
    dest: "{{ item }}"
    state: directory
    mode: "750"
  with_items:
    - /etc/firewalld
    - /etc/firewalld/zones
  notify: firewalld__reload
```

It is not correct that "750" is mis-interpreted. "750" is a string, and is correctly interpreted as the octal permission number `750` i.e. `rwxr-x---`.

Personally I have been using the string "750", rather than remember that YAML supports octal numeric literals using a leading zero.

**Comment by awcrosby** [comment]
Rule 202 should be edited to check for quotes or a leading zero, as specified in the file module doc. Thanks for the catch!
https://github.com/ansible/ansible-lint/blob/991079d5a865c5fb4a6dd05242b5058cb57bff41/lib/ansiblelint/rules/OctalPermissionsRule.py#L26-L33
